[hotfix] [rat] Add exclusion for test state snapshots

zentol opened the pull request:
Fixes the compilation on Windows by adding rat exclusions for recently added state snapshots.

This must be merged to 1.3 and 1.4.

StephanEwen commented:
Can we use a standardized prefix / suffix for all snapshot files and have one exclusion line in?
The exclusions will grow crazy over time otherwise...

zentol commented:
We could do that. These files shouldn't care what their name is, so standardizing them should be rather easy.

zentol commented:
I'll look into renaming the files.

zentol commented:
I've renamed all snapshots/savepoints that I could find. I've added exclusion rules for files ending in `snapshot` and `savepoint`.

rat-plugin passed locally.

tzulitai commented on flink-tests/src/test/java/org/apache/flink/test/checkpointing/utils/StatefulJobSavepointFrom11MigrationITCase.java:
```diff
@@ -129,7 +129,7 @@ public void testCreateSavepointOnFlink11WithRocksDB() throws Exception {
 		executeAndSavepoint(
 				env,
-				"src/test/resources/stateful-udf-migration-itcase-flink1.1-savepoint-rocksdb",
+				"src/test/resources/stateful-udf-migration-itcase-flink1.1-rocksdbsavepoint",
```
I this supposed to be `rocksdb-savepoint`? Otherwise shouldn't be detected by `*-savepoint`.

tzulitai commented:
Looks good to me, minus one minor comment.

zentol commented:
@tzulitai Nice catch, will fix it while merging.
